src/zenml/stack/stack.py

```diff
@@ -481,7 +482,9 @@ def dict(self) -> Dict[str, str]:
             A dictionary containing the stack components.
         """
         component_dict = {
-            component_type.value: component.config.json(sort_keys=True)
+            component_type.value: json.dumps(
+                component.config.model_dump(mode="json"), sort_keys=True
```

Comment:
why not `model_dump_json`?

Reply:
This has a silly reason. Back in V1, you could call `.json()` on a pydantic model and use the parameter `sort_keys`. However, this option does not exist in the V2-equivalent function `model_dump_json`. To keep the behavior consistent, I had to jump through a few hoops.